corehq/apps/data_interfaces/views.py

```diff
@@ -531,10 +521,11 @@ def inaccessible_forms_accessed(self, xform_ids, domain, couch_user):
         return xforms_user_ids - accessible_user_ids
 
     def get_xform_ids(self, request):
-        form_ids_or_query_string = self.get_form_ids_or_query_string(request)
-        if type(form_ids_or_query_string) == list:
-            xform_ids = form_ids_or_query_string
-        elif isinstance(form_ids_or_query_string, basestring):
+        if 'select_all' in self.request.POST:
+            # Altough evaluating form_ids and sending to task would be cleaner,
+            # heavier calls should be in an async task instead
```

Comment:
hmm, I just took a closer look at what this existing code (and this comment) were doing - the comment refers to behavior that was just changed in the last PR (the "heavier call" no longer happens in the async task).

Also, it's out of scope to change, but that request mocking business seems like a needlessly complex way of performing an elasticsearch query.